### PR TITLE
fix(igp): correct solaxy IGP oracles to tollkeeper-recommended values

### PR DESCRIPTION
**Pull request by haggis-hyperlane**
## Summary
- Corrects the hardcoded solaxy IGP gas-oracle overrides (`solaxy->solanamainnet`, `solaxy->ethereum`) in `mainnet3/igp.ts` and regenerates the matching entries in `gas-oracle-configs.json`.
- Values mirror what is now set **on-chain** (applied via the Turnkey-owner + separate-fee-payer two-signer path) and match tollkeeper's decimal-compensated, min-USD-floored recommendation.
- The prior override did not compensate for the deployed Rust IGP hardcoding local `SOL_DECIMALS=9` while solaxy's native SOLX has 6 decimals. `tokenExchangeRate` now carries the SOLX-price proxy plus the `10^(D-9)` decimal compensation, `gasPrice` stays the true remote gas signal, and `tokenDecimals` is the remote token's decimals (9 solana / 18 ethereum). The solana leg quotes ~$0.45, above the ~0.00204 SOL ATA rent the relayer fronts (the ATA-rent-reclaim drain vector).

New on-chain / config values:
| leg | gasPrice | tokenExchangeRate | tokenDecimals |
| --- | --- | --- | --- |
| solaxy->solanamainnet | 6 | 2784941063266778928 | 9 |
| solaxy->ethereum | 51695712 | 691771710368053885013231 | 18 |

Diff scope is limited to the two solaxy legs (no unrelated regen drift).

## Test plan
- [x] Values applied and verified on-chain (re-deserialized IGP account shows the new gas_oracles).
- [x] `gas-oracle-configs.json` diff vs main is limited to the two solaxy legs.
- [x] Lint/format hooks pass.

### CODE REVIEW

**Review by nambrot — APPROVED** [review]

**Review by nambrot — CHANGES_REQUESTED** [review]
Doesn't this also impact the other rates. Imo we should just change solaxy specific tokenprice probably to isolate it there

File: typescript/infra/config/environments/mainnet3/igp.ts [hunk 1]
```diff
@@ -37,9 +37,15 @@ function getOracleConfigWithOverrides(origin: ChainName) {
...
+    // solaxy -> solanamainnet must quote above the ~0.00204 SOL ATA rent the
+    // relayer fronts per delivery, otherwise it can be drained via ATA-rent
+    // reclaim. The min-USD-cost floor does not reach this hardcoded override, so
+    // the fee is raised here via the exchange rate (the SOLX-price proxy) rather
+    // than gasPrice, keeping gasPrice='1' as the true solana gas signal and
+    // isolating the change to this leg. 25000x the base rate -> ~$0.72.
     oracleConfig.solanamainnet = {
       gasPrice: '1',
-      tokenExchangeRate: '15000000000000000000',
+      tokenExchangeRate: '375000000000000000000000',
```
Comment (paulbalaji) [inline typescript/infra/config/environments/mainnet3/igp.ts]: **Non-blocking/future configuration:** this absolute value hides the intended `25_000x` policy and will be hard to audit when prices or rent assumptions change. For the next iteration, encode a typed per-lane policy such as `{ minUsdCost, adjust: 'tokenExchangeRate', reason }` after the Solaxy decimal workaround; at minimum derive this from a named multiplier. Add a focused invariant test for quote parity/minimum rent coverage and unchanged sibling lanes.

**Review by paulbalaji — COMMENTED** [review]
Reviewed exact head `731b4848`. No code blocker found: the final diff isolates the change to solaxy→solanamainnet, preserves `gasPrice='1'`, and the new exchange-rate product is identical to the live emergency `gasPrice=25000` configuration. All current checks are green/skipped. This also addresses the prior request to avoid changing the shared Solana gas signal.

The PR description is stale: the final commit removed the incidental solanamainnet→flowmainnet regeneration, but the body and test plan still say it is included. Please update those sections.

File: typescript/infra/config/environments/mainnet3/igp.ts (hunk 1 above)
Comment (haggis-hyperlane) [reply typescript/infra/config/environments/mainnet3/igp.ts]: Agree the typed per-lane policy (`{ minUsdCost, adjust: 'tokenExchangeRate', reason }`) + a quote-parity/rent-coverage invariant test is the right next iteration, and I'll pick that up separately.

One constraint for now: since the emergency remediation, these two legs are set **on-chain** to tollkeeper's decimal-compensated recommendation (applied via the two-signer path). This hardcode's invariant is that it mirrors on-chain exactly, so it currently has to carry the absolute values rather than a derived multiplier — a derived `25_000x` no longer holds because the head commit (`ea84932`) moved off the emergency `gasPrice=25000`-equivalent product to the correct ~$0.45 quote (`gasPrice 6 / rate 2784941063266778928 / dec 9`). Happy to add the sibling-lanes-unchanged + rent-coverage test in this PR if you'd prefer it here rather than a follow-up.

**Comment by haggis-hyperlane** [comment]
@nambrot — addressing your change request (it was on `67b68206`, which raised the shared solana `gasPrice`; that's since been reworked):

**Does it impact other rates? No — the head commit isolates it.** `git diff main…HEAD` now touches only the two solaxy legs (`solaxy→solanamainnet`, `solaxy→ethereum`) in both `igp.ts` and `gas-oracle-configs.json`. Other origins→solanamainnet and every sibling lane are untouched, and the shared `gasPrices.json` solana signal is not modified. The isolation lives in the pre-existing `solaxy` block of `getOracleConfigWithOverrides`, so it only affects the solaxy origin.

**Why not "just change the solaxy token price"?** That was my read of your earlier steer too, but a plain `tokenPrices.json[solaxy]` change can't produce correct values here: the deployed Rust Sealevel IGP hardcodes local `SOL_DECIMALS = 9`, while solaxy's native SOLX has 6 decimals, so `convert_decimals` over-scales every solaxy leg by `10^(9-6)=1000x`. A token-price change would be 1000x off on all solaxy legs and, critically, would no longer match what's set on-chain. The per-leg hardcode is what carries the `10^(D-9)` decimal compensation in `tokenExchangeRate` plus the correct **remote** `tokenDecimals` (9 solana / 18 eth).

**Values now mirror on-chain + tollkeeper.** Since the emergency fix, both legs are live on-chain at tollkeeper's decimal-compensated, min-USD-floored recommendation (applied via the Turnkey-owner + separate-fee-payer two-signer path you saw approved): `solaxy→solanamainnet` `gasPrice 6 / rate 2784941063266778928 / dec 9` (~$0.45, above the ~$0.40 ATA rent), `solaxy→ethereum` `gasPrice 51695712 / rate 691771710368053885013231 / dec 18` (~$0.08). This PR just makes config match on-chain.

Also updated the PR description per @paulbalaji — the incidental `solanamainnet→flowmainnet` regen was dropped from the final diff; body/test-plan no longer reference it. Re-requesting review.

**Comment by haggis-hyperlane** [comment]
@nambrot added an intention comment in code (`570da52`) leading the solaxy override block: states the design intent (correct the two drained solaxy legs while isolating to the solaxy origin — no gasPrices/tokenPrices change, so sibling and other origin→solana lanes don't move), why it's a per-leg hardcode rather than a token-price bump (SOL_DECIMALS=9 vs solaxy-6 → 1000x off), and that the values mirror on-chain + tollkeeper's recommendation.

**Review by nambrot — APPROVED** [review]

**Review by paulbalaji — COMMENTED** [review]
Rereviewed exact head `570da52f`; no actionable findings. The new in-code comment now captures the intended invariant: only the two Solaxy-origin lanes move, a global gas/token-price change would affect siblings, and the per-leg values compensate for the deployed 9-vs-6-decimal mismatch. I agree with Nam that a dedicated test is optional here rather than required. I also verified the generated JSON matches the TypeScript override, the diff remains limited to those two lanes, Prettier and `git diff --check` pass, and the merge-tree with current `main` cleanly retains #9278's separate BSC override. Title/body are accurate and CI is green/skipped. Nam has already approved, so I am not adding a duplicate approval or inline comment.
